datalad/interface/tests/test_run_procedure.py

```diff
@@ -304,17 +305,52 @@ def test_quoting(path):
         with assert_raises(CommandError):
             runner.run("datalad run-procedure just2args 'still-one arg'")
 
+
 @skip_if_on_windows
-@with_tempfile
-def test_text2git_empty(path):
-    """
-    Tests that empty files are not annexed in a ds configured with text2git.
-    """
+@with_tree(tree={
+    # "TEXT" ones
+    'empty': '',  # we have special rule to treat empty ones as text
+    # check various structured files - libmagic might change its decisions which
+    # can effect git-annex. https://github.com/datalad/datalad/issues/3361
+    'JSON': """\
+{
+    "name": "John Smith",
+    "age": 33
+}
+""",
+    'YAML': """\
+--- # The Smiths
+- {name: John Smith, age: 33}
+- name: Mary Smith
+  age: 27
+""",
+    'MARKDOWN': """\
+# Title
+
+## Section1
+
+When the earth was flat
+
+## Section2
+""",
+    # BINARY ones
+    '0blob': '\x00',
+    'emptyline': '\n',  # libmagic: "binary" "application/octet-stream"
+})
+def test_text2git(path):
+    # Test if files being correctly annexed in a ds configured with text2git.
+    TEXT_FILES = ('JSON', 'YAML', 'MARKDOWN', 'empty')
+    BINARY_FILES = ('0blob', 'emptyline')
+
     ds = Dataset(path).create(force=True)
     ds.run_procedure('cfg_text2git')
+
+    ds.save(path=TEXT_FILES + BINARY_FILES, message="added all files")
     assert_repo_status(ds.path)
-    # create an empty file, no extension
-    open(op.join(path, 'emptyfile'), 'a').close()
-    ds.save(message="add empty file")
-    # check that it's not annexed
-    assert_false(ds.repo.is_under_annex("emptyfile"))
+
+    # check that text files are not annexed
+    for f in TEXT_FILES:
+        assert_false(ds.repo.is_under_annex(f))
+    # and trivial binaries - annexed
+    for f in BINARY_FILES:
+        assert_true(ds.repo.is_under_annex(f))
```

Comment:
Hmm, da27e86c4 (part of gh-3956) dropped the `assert_false(ds.repo.is_under_annex("emptyfile"))` above for these `ok_file_under_git` calls.  Why aren't you keeping that change here?

Reply:
this whole test was pretty much rewritten from being a test for only an empty file, to a test which tests various types (text and "binary") of files -- from `test_text2git_empty` to `test_text2git`. So `diff` here could be a bit misleading - I would not strive to minimize diff here, nor re-adding `assert_false(ds.repo.is_under_annex("emptyfile"))` would add any additional testing

Reply:
Let me try again.  Here is the resolution that you pushed:

```
+     ds.save(path=TEXT_FILES + BINARY_FILES, message="added all files")
 -    ok_clean_git(ds.path)
 +    assert_repo_status(ds.path)
-     # create an empty file, no extension
-     open(op.join(path, 'emptyfile'), 'a').close()
-     ds.save(message="add empty file")
-     # check that it's not annexed
-     assert_false(ds.repo.is_under_annex("emptyfile"))
+ 
+     # check that text files are not annexed
+     for f in TEXT_FILES:
 -        ok_file_under_git(path, f, annexed=False)
++        assert_false(ds.repo.is_under_annex(f))
+     # and trivial binaries - annexed
+     for f in BINARY_FILES:
 -        ok_file_under_git(path, f, annexed=True)
++        assert_true(ds.repo.is_under_annex(f))
```

Note that you're introducing two changes (the `++` liines) that come from neither side.  Unless there is a master-specific reason for why doing that, you should not.

Reply:
hm, IIRC I was just replacing "0.11.x" style (`ok_file_under_git(path, f, annexed=False)`) from one side to "master style" (`assert_false(ds.repo.is_under_annex(f))`).  What should have I done differently?

Reply:
> I was just replacing "0.11.x" style (`ok_file_under_git(path, f, annexed=False)`) from one side to "master style" (`assert_false(ds.repo.is_under_annex(f))`)

Thanks, knowing your intention here helps.

Given that the `assert_false(ds.repo.is_under_annex(f))` line was shared with 0.11.x, and you just did the reverse substitution in 0.11.x's da27e86, in my view the commit message should explain why we should prefer one form on master and the other on 0.11.x, ideally with a pointer to where we've discussed `ok_file_under_git` being problematic.  The main thing I can find is gh-2961, but it looks like that was resolved by your gh-2972.

Even more ideally, if `ok_file_under_git` is still an issue and should be avoided on master, we'd also avoid it on 0.11.x, because (1) it seems likely that the same issues would be relevant for 0.11.x and (2) it introduces needless differences when the master variant will work fine on 0.11.x.

I don't think it is worth a PR to 0.11.x at this point, so I'll go with adding something to the commit message to the merge that I force-push here.

Reply:
thanks for the analysis! I just saw difference, so assumed that it would be the right way to do in master without checking compatibility with 0.11.x .  Please do as you see it fit! Thanks in advance!